core/connection/manager.go

```diff
@@ -450,12 +429,17 @@ func (manager *connectionManager) Disconnect() error {
 	}
 
 	manager.setStatus(statusDisconnecting())
-	manager.cleanConnection()
-	manager.setStatus(statusNotConnected())
-	manager.publishStateEvent(NotConnected)
+	manager.cancel()
+
+	log.Debug().Msg("Connection stopping..")
+	manager.connection.Stop()
+	<-manager.connectionStopped
```

Comment:
`Disconnect` can be called from many places, but `connectionStopped` closed only in the `consumeConnectionStates`. 
If `Disconnect` was called before `consumeConnectionStates` it will never end. 

Isn't it enough to just call `manager.connection.Stop()` to make sure that connection stopped?

Reply:
>> If Disconnect was called before consumeConnectionStates it will never end.

Disconnect handles that and does not start it's logic if connection is not created:
```
if manager.Status().State == NotConnected {
		return ErrNoConnection
	}
```

>> Isn't it enough to just call manager.connection.Stop() to make sure that connection stopped?
No, that's what this PR adds - waiting for actual stop e.g. PROCESS_EXITED event in Openvpn case

Reply:
It only checks for `NotConnected` state, but what about others?

> No, that's what this PR adds - waiting for actual stop e.g. PROCESS_EXITED event in Openvpn case

Shouldn't it be a part of the connection.Stop()?